include/quda.h

```diff
@@ -814,8 +870,28 @@ extern "C" {
    * @param param  Contains all metadata regarding host and device
    *               storage and solver parameters
    */
-  void lanczosQuda(int k0, int m, void *hp_Apsi, void *hp_r, void *hp_V,
-                   void *hp_alpha, void *hp_beta, QudaEigParam *eig_param);
+  void lanczosQuda(int k0, int m, void *hp_Apsi, void *hp_r, void *hp_V, void *hp_alpha, void *hp_beta,
+                   QudaEigParam *eig_param);
+
+  /**
+   * Perform the eigensolve. The problem matrix is defined by the invert param, the
+   * mode of solution is specified by the eig param. It is assumed that the gauge
+   * field has already been loaded via  loadGaugeQuda().
+   * @param h_els  Host side eigenvalues
+   * @param h_evs  Host side eigenvectors
+   * @param param  Contains all metadata regarding the type of solve.
+   */
+  void eigensolveQuda(void *h_evals, void *h_evecs, QudaEigParam *param);
+
+  /**
+   * Perform the eigensolve using ARPACK. The problem matrix is defined by the
+   * invert param, the mode of solution is specified by the eig param.
+   * It is assumed that the gauge field has already been loaded via loadGaugeQuda().
+   * @param h_els  Host side eigenvalues
+   * @param h_evs  Host side eigenvectors
+   * @param param  Contains all metadata regarding the type of solve.
+   */
+  void eigensolveARPACK(void *h_evals, void *h_evecs, QudaEigParam *param);
```

Comment:
If this is a public function to be called from applications then this should be `eigensolveARPACKQuda` to follow the usual naming convention.  Perhaps it should just be subsumed into `eigensolveQuda` though as an option to use `ARPACK` instead of native QUDA?